### ISSUE TITLE
Generated relations contains only one entry

### ISSUE THREAD

**Issue by ebenmoha** [strongloop/loopback#3050]
The discoverSchema method returns only 1 entry for the relations while I have multiple relations for each table. I tried to manually add them, but the "discoverForeignKeys" method is returning wrong values for the table that references the association. Here's an example: 

{ fkOwner: 'dbo',
    fkName: 'FK_Foo_Bar',
    fkTableName: 'Foo',
    fkColumnName: 'Foocolumn',
    keySeq: 1,
    pkOwner: 'dbo',
    pkName: 'PK',
    pkTableName: 'Foo',
    pkColumnName: 'Foocolumn' },

In here there's a "HasOne" relation between Foo and Bar, but the result in fkTableName and pkTableName is the same. Also, the generated relation by "discoverSchema" method  is always the last relation. So, I think that, the bug is only overriding the relations key. As the relations key is an object not an array. Thanks for your help.

**Comment by jannyHou** [comment]
Hey @ebenmoha I am trying to reproduce your problem, which connector are you using?

**Comment by ebenmoha** [comment]
Hello @jannyHou . I am using the mssql connector, the connector and its juggler doesn't seem to support many things like PK identity, hasOne, hasMany, and hasManyThrough relations. I am actually working on a fork of this two libraries (connector + juggler) so that it can handle all this things. I created a first pull request for loopback-connector-mssql that fixes this bug and also a multiple primary key issue. 

In addition, locally, I added the identity, hasOne and hasMany relations features and will push them to my fork soon.
